File: architecturecomponents/src/main/java/com/fesskiev/architecturecomponents/di/DataSourceModule.java
package com.fesskiev.architecturecomponents.di;


import android.content.Context;

import com.fesskiev.architecturecomponents.domain.source.local.LocalDataSource;
import com.fesskiev.architecturecomponents.domain.source.local.room.WeatherDatabase;
import com.fesskiev.architecturecomponents.domain.source.remote.RemoteDataSource;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import retrofit2.Retrofit;

@Module
public class DataSourceModule {

    @Provides
    @Singleton
    public RemoteDataSource provideRemoteDataSource(Context context, Retrofit retrofit) {
        return new RemoteDataSource(context, retrofit);
    }

    @Provides
    @Singleton
    public LocalDataSource provideLocalDataSource(WeatherDatabase database) {
        return new LocalDataSource(database);
    }

}

File: graphql/src/main/kotlin/com/fesskiev/graphql/di/DataSourceModule.kt
package com.fesskiev.graphql.di

import android.content.Context
import com.fesskiev.graphql.domain.source.DataRepository
import com.fesskiev.graphql.domain.source.remote.RemoteDataSource
import dagger.Module
import dagger.Provides
import retrofit2.Retrofit
import javax.inject.Singleton

@Module
class DataSourceModule {

    @Provides
    @Singleton
    fun provideRemoteDataSource(context: Context, retrofit: Retrofit): RemoteDataSource {
        return RemoteDataSource(context, retrofit)
    }

    @Provides
    @Singleton
    fun provideRepository(remoteDataSource: RemoteDataSource): DataRepository {
        return DataRepository(remoteDataSource)
    }
}